snapshot/cli/cli.go

```diff
@@ -0,0 +1,84 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/scootdev/scoot/snapshot"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+type DBInjector interface {
+	RegisterFlags(rootCmd *cobra.Command)
+	Inject() (snapshot.DB, error)
+}
+
+func MakeDBCLI(injector DBInjector) *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:   "scoot-snapshot-db",
+		Short: "scoot snapshot db CLI",
+	}
+	parentCmd := rootCmd
+	add := func(subCmd dbCommand) {
+		cmd := subCmd.register()
+		cmd.RunE = func(innerCmd *cobra.Command, args []string) error {
+			db, err := injector.Inject()
+			if err != nil {
+				return fmt.Errorf("scoot-snapshot-db could not create db: %v", err)
+			}
+			return subCmd.run(db, innerCmd, args)
+		}
+		parentCmd.AddCommand(cmd)
+	}
+
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "create a snapshot",
+	}
+	rootCmd.AddCommand(createCmd)
+	parentCmd = createCmd
+
+	add(&ingestGitCommitCommand{})
```

Comment:
Add() should probably take cobra parentCmd as a param and return the new cobra cmd.